provider/spark.go

```diff
@@ -911,6 +998,62 @@ func (e *EMRExecutor) SparkSubmitArgs(destPath string, cleanQuery string, source
 	return argList
 }
 
+func createLogS3FileStore(emrRegion string, s3LogLocation string, awsAccessKeyId string, awsSecretKey string) (FileStore, error) {
+	if s3LogLocation == "" {
+		return nil, nil
+	}
+	bucketName, path := getBucketAndPathFromFilePath(s3LogLocation)
+
+	logS3Config := pc.S3FileStoreConfig{
+		Credentials:  pc.AWSCredentials{AWSAccessKeyId: awsAccessKeyId, AWSSecretKey: awsSecretKey},
+		BucketRegion: emrRegion,
+		BucketPath:   bucketName,
+		Path:         path,
+	}
+
+	config, err := logS3Config.Serialize()
+	if err != nil {
+		return nil, fmt.Errorf("could not serialize s3 file store config: %v", err)
+	}
+
+	logFileStore, err := NewS3FileStore(config)
+	if err != nil {
+		return nil, fmt.Errorf("could not create s3 file store (bucket: %s, path: %s) for emr logs: %v", err, bucketName, path)
+	}
+	return logFileStore, nil
+}
+
+func getBucketAndPathFromFilePath(filePath string) (string, string) {
+	logFileWithoutPrefix := strings.Split(filePath, "://")[1]
+	endOfBucketNameIdx := strings.Index(logFileWithoutPrefix, "/")
+	bucket := logFileWithoutPrefix[:endOfBucketNameIdx]
+	path := strings.Trim(logFileWithoutPrefix[endOfBucketNameIdx:], "/")
+
+	return bucket, path
+}
+
+func decompressMessageToString(message []byte) (string, error) {
```

Comment:
Can we move this to its own package? Seems generic enough

Reply:
does it make sense to move it to helpers?